fe/fe-core/src/main/java/com/starrocks/analysis/StringLiteral.java

```diff
@@ -66,7 +66,9 @@ public int compareLiteral(LiteralExpr expr) {
         if (expr instanceof NullLiteral) {
             return 1;
         }
-
+        if (expr == MaxLiteral.MAX_VALUE) {
```

Comment:
use equals

Reply:
MaxLiteral.MAX_VALUE is singleton，it can use == compare directly